src/controllers/projects.js

```diff
@@ -1,189 +1,71 @@
-const fs = require('fs');
-const projects = require('../data/projects.json');
-const employeesData = require('../data/employees.json');
+import Projects from '../models/Projects';
 
-// getAll projects
-const getAllProjects = (req, res) => {
-  res.status(200).json(projects);
-};
-
-// getActive projects
-const getActiveProjects = (req, res) => {
-  const foundProjects = projects.filter((project) => project.isDeleted === false);
-  res.status(200).json(foundProjects);
-};
+// GetAll
+const getAll = async (req, res) => {
+  try {
+    const projects = await Projects.find();
 
-// getById projects
-const getProjectById = (req, res) => {
-  const projectId = parseInt(req.params.id, 10);
-  const foundProject = projects.find((project) => project.id === projectId);
-  if (foundProject) {
-    res.status(200).json(foundProject);
-  } else {
-    res.status(404).json({ msg: `Project not found by id: ${projectId}`, error: true });
-  }
-};
-
-// create projects
-const createProject = (req, res) => {
-  const projectId = Number(req.body.id);
-  const oneProject = projects.find((project) => project.id === projectId);
-  const newProject = req.body;
-  const saveProject = {
-    id: newProject.id = Number(new Date().getTime().toString().substring(6)),
-    name: newProject.name,
-    lastName: newProject.lastName,
-    email: newProject.email,
-    password: newProject.password,
-  };
-  if (JSON.stringify(newProject) === '{}') {
-    res.status(400).send('New Project is empty');
-  } else if (oneProject) {
-    res.status(400).json({ msg: 'Project already exists!' });
-  } else {
-    projects.push(saveProject);
+    return res.status(200).json({
+      message: 'All Projects',
+      data: projects,
+      error: false,
+    });
+  } catch (err) {
+    return res.json({
+      message: 'An error occurred',
+      error: err,
+    });
   }
-  fs.writeFile('src/data/projects.json', JSON.stringify(projects), (err) => {
-    if (err) {
-      res.send('Cannot save New Project');
-    } else {
-      res.status(201).json('Project Created');
-    }
-  });
 };
 
-// filter projects
-const filterProjects = (req, res) => {
-  let projectArray = projects;
-  const queriesArray = Object.keys(req.query);
-  queriesArray.forEach((query) => {
-    if (query !== 'id' && query !== 'name' && query !== 'startDate' && query !== 'endDate' && query !== 'clientName' && query !== 'employees') {
-      res.status(400).json({
-        Error: 'Applied filter does not exist',
-      });
-    }
-  });
+// GetById
+const getById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const projects = await Projects.findById(id);
```

Comment:
```suggestion
    const project = await Projects.findById(id);
```
The findById method could return a null/undefined value. So you have to add an extra validation and check the value of the project variable.
If it's undefined you have to return a different response.